Crash with Error: socket hang up

**Describe the bug**
One of our services has started crashing, and the last thing printed out  is the following:
```
 Error: socket hang up
     at createHangUpError (_http_client.js:332:15)
     at Socket.socketCloseListener (_http_client.js:373:25)
     at scope.activate (/apps/ts/node_modules/dd-trace/packages/dd-trace/src/scope/base.js:54:19)
     at Scope._activate (/apps/ts/node_modules/dd-trace/packages/dd-trace/src/scope/async_hooks.js:54:14)
     at Scope.activate (/apps/ts/node_modules/dd-trace/packages/dd-trace/src/scope/base.js:12:19)
     at Socket.bound (/apps/ts/node_modules/dd-trace/packages/dd-trace/src/scope/base.js:53:20)
     at Socket.emit (events.js:203:15)
     at Socket.EventEmitter.emit (domain.js:448:20)
     at TCP._handle.close (net.js:607:12)
```
If I disable tracing in the init via enabled:false, then it stops crashing, we didn't see this on older versions of dd-trace.

The exit status we get is `1`

**Environment**

* **Operation system:**
Ubuntu 20.04
* **Node version:**
v10.19.0
* **Tracer version:**
v0.30.6
* **Agent version:**
Agent (v7.24.1)

Had the same result using an Axios call as @trajanoreuter brought up

Agent: 6.24.1
dd-trace: 0.28.0
Axios: 0.20.0

`timeout of 50ms exceeded`

```
/server/node_modules/dd-trace/packages/dd-trace/src/scope/base.js:18
throw e
^
Error: socket hang up
at connResetException (internal/errors.js:610:14)
at TLSSocket.socketCloseListener (_http_client.js:400:25)
at /server/node_modules/dd-trace/packages/dd-trace/src/scope/base.js:54:19
at Scope._activate (/server/node_modules/dd-trace/packages/dd-trace/src/scope/async_hooks.js:54:14)
at Scope.activate (/server/node_modules/dd-trace/packages/dd-trace/src/scope/base.js:12:19)
at TLSSocket.bound (/server/node_modules/dd-trace/packages/dd-trace/src/scope/base.js:53:20)
at TLSSocket.emit (events.js:327:22)
at net.js:674:12
at TCP.done (_tls_wrap.js:567:7)
at TCP.callbackTrampoline (internal/async_hooks.js:120:14)
Emitted 'error' event on Writable instance at:
at ClientRequest.eventHandlers.<computed> (/server/node_modules/follow-redirects/index.js:13:24)
at /server/node_modules/dd-trace/packages/dd-trace/src/scope/base.js:54:19
at Scope._activate (/server/node_modules/dd-trace/packages/dd-trace/src/scope/async_hooks.js:54:14)
at Scope.activate (/server/node_modules/dd-trace/packages/dd-trace/src/scope/base.js:12:19)
at ClientRequest.bound (/server/node_modules/dd-trace/packages/dd-trace/src/scope/base.js:53:20)
at ClientRequest.emit (events.js:327:22)
at ClientRequest.req.emit (/server/node_modules/dd-trace/packages/datadog-plugin-http/src/client.js:106:21)
at TLSSocket.socketCloseListener (_http_client.js:400:11)
at /server/node_modules/dd-trace/packages/dd-trace/src/scope/base.js:54:19
at Scope._activate (/server/node_modules/dd-trace/packages/dd-trace/src/scope/async_hooks.js:54:14) {
code: 'ECONNRESET'
}
2021-03-04T16:35:58.397-03:00	npm ERR! code ELIFECYCLE
2021-03-04T16:35:58.397-03:00	npm ERR! errno 1
```

> Had the same result using an Axios call as @trajanoreuter brought up
> 
> Agent: 6.24.1
> dd-trace: 0.28.0
> Axios: 0.20.0
> 
> `timeout of 50ms exceeded`
> 
> ```
> /server/node_modules/dd-trace/packages/dd-trace/src/scope/base.js:18
> throw e
> ^
> Error: socket hang up
> at connResetException (internal/errors.js:610:14)
> at TLSSocket.socketCloseListener (_http_client.js:400:25)
> at /server/node_modules/dd-trace/packages/dd-trace/src/scope/base.js:54:19
> at Scope._activate (/server/node_modules/dd-trace/packages/dd-trace/src/scope/async_hooks.js:54:14)
> at Scope.activate (/server/node_modules/dd-trace/packages/dd-trace/src/scope/base.js:12:19)
> at TLSSocket.bound (/server/node_modules/dd-trace/packages/dd-trace/src/scope/base.js:53:20)
> at TLSSocket.emit (events.js:327:22)
> at net.js:674:12
> at TCP.done (_tls_wrap.js:567:7)
> at TCP.callbackTrampoline (internal/async_hooks.js:120:14)
> Emitted 'error' event on Writable instance at:
> at ClientRequest.eventHandlers.<computed> (/server/node_modules/follow-redirects/index.js:13:24)
> at /server/node_modules/dd-trace/packages/dd-trace/src/scope/base.js:54:19
> at Scope._activate (/server/node_modules/dd-trace/packages/dd-trace/src/scope/async_hooks.js:54:14)
> at Scope.activate (/server/node_modules/dd-trace/packages/dd-trace/src/scope/base.js:12:19)
> at ClientRequest.bound (/server/node_modules/dd-trace/packages/dd-trace/src/scope/base.js:53:20)
> at ClientRequest.emit (events.js:327:22)
> at ClientRequest.req.emit (/server/node_modules/dd-trace/packages/datadog-plugin-http/src/client.js:106:21)
> at TLSSocket.socketCloseListener (_http_client.js:400:11)
> at /server/node_modules/dd-trace/packages/dd-trace/src/scope/base.js:54:19
> at Scope._activate (/server/node_modules/dd-trace/packages/dd-trace/src/scope/async_hooks.js:54:14) {
> code: 'ECONNRESET'
> }
> 2021-03-04T16:35:58.397-03:00	npm ERR! code ELIFECYCLE
> 2021-03-04T16:35:58.397-03:00	npm ERR! errno 1
> ```

This is almost identical to the issue we have encountered.

--- a/packages/dd-trace/src/scope/base.js
+++ b/packages/dd-trace/src/scope/base.js
@@ -188,7 +188,7 @@ function wrapRemoveListener (removeListener) {
 
     listeners.delete(listener)
 
-    return removeListener.call(this, eventName, listener)
+    return removeListener.apply(this, arguments)
   }
 }
 
@@ -202,7 +202,7 @@ function wrapRemoveAllListeners (removeAllListeners) {
       }
     }
 
-    return removeAllListeners.call(this, eventName)
+    return removeAllListeners.apply(this, arguments)
   }
 }
 

--- a/packages/dd-trace/test/scope/test.js
+++ b/packages/dd-trace/test/scope/test.js
@@ -300,6 +300,43 @@ module.exports = factory => {
 
         expect(spy).to.not.have.been.called
       })
+
+      it('should remove all listeners of a type', () => {
+        const spy = sinon.spy()
+        const spy2 = sinon.spy()
+
+        scope.bind(emitter)
+
+        scope.activate(span, () => {
+          emitter.once('test', spy)
+          emitter.once('test', spy2)
+          emitter.removeAllListeners('test')
+        })
+
+        emitter.emit('test')
+
+        expect(spy).to.not.have.been.called
+        expect(spy2).to.not.have.been.called
+      })
+
+      it('should remove all listeners', () => {
+        const spy = sinon.spy()
+        const spy2 = sinon.spy()
+
+        scope.bind(emitter)
+
+        scope.activate(span, () => {
+          emitter.once('test', spy)
+          emitter.once('test2', spy2)
+          emitter.removeAllListeners()
+        })
+
+        emitter.emit('test')
+        emitter.emit('test2')
+
+        expect(spy).to.not.have.been.called
+        expect(spy2).to.not.have.been.called
+      })
     })
 
     describe('with an unsupported target', () => {

